src/state.ts

```diff
@@ -253,17 +253,18 @@ async function findProjectRootUri(projectFileNames, doc, workspaceFolder): Promi
     let searchUri = doc?.uri || workspaceFolder?.uri;
     let prev = null;
     while (searchUri != prev) {
-        for (let projectFile in projectFileNames) {
-            const u = vscode.Uri.joinPath(searchUri, projectFileNames[projectFile]);
-            try {
-                const stat = await vscode.workspace.fs.stat(u);
-                if (stat) {
+        try {
+            for (let projectFile in projectFileNames) {
+                const u = vscode.Uri.joinPath(searchUri, projectFileNames[projectFile]);
+                try {
+                    await vscode.workspace.fs.stat(u);
                     cursor.set(PROJECT_DIR_URI_KEY, searchUri);
                     return;
                 }
+                catch { }
             }
-            catch (_) { }
         }
+        catch (_) { }
```

Comment:
Should we log something useful here in case we have an issue later, so we can easily find the source?

Reply:
Yes we should. Good catch!